internal/insights/store/store.go

```diff
@@ -976,11 +972,34 @@ func (s *Store) GetAllDataForInsightViewID(ctx context.Context, opts ExportOpts)
 		return nil, errors.Wrap(err, "fetching code insights data")
 	}
 
+	results = FilterSeriesPoints(denylist, results)
+
 	return results, nil
 }
 
+func FilterSeriesPoints(denyList []api.RepoID, array []SeriesPointForExport) []SeriesPointForExport {
+	// If there is nothing to filter, then return early. This skips the assignment in the later for-loop.
+	if len(denyList) == 0 {
+		return array
+	}
+	// We turn the denyList into a map to ensure O(n) time complexity
+	denyMap := make(map[int32]bool)
```

Comment:
This can be pre-allocated to avoid making extra allocations when it grows below

```suggestion
	denyMap := make(map[int32]bool, len(denyList))
```

Reply:
Done!